core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -1647,6 +1645,24 @@ class SnappySession(@transient private val sc: SparkContext,
 
   def setPreparedQuery(preparePhase: Boolean, paramSet: Option[ParameterValueSet]): Unit =
     sessionState.setPreparedQuery(preparePhase, paramSet)
+
+  /**
+   * Callback to execute code when a particular property is set.
+   * @param key
+   * @param value
+   */
+  def confCallback(key: String, value: String): Unit = {
+    key match {
+      case "spark.scheduler.pool" =>
+        // set the scheduler pool if the pool exists else throw an exception. 
+        if (sparkContext.getAllPools.exists(_.name == value)) {
+          sparkContext.setLocalProperty(key, value)
+        } else {
+          throw new IllegalArgumentException(s"Invalid Pool $value")
+        }
+      case _ =>
+    }
+  }
```

Comment:
Yeah. But given that it is spark app, it would be better to leave it the spark way. Setting scheduler pools for embedded mode and smart connector in the same way (which is different than spark) can confuse users. In fact, I wanted to change the property name from spark.scheduler.pool to a snappy specific name, say snappydata.execution.pool, for embedded mode. This would avoid any confusion related to spark app. I discussed this with @jramnara and he was ok with spark.scheduler.pool so I left it that way. What do you say?